rsrc/rsrc.go

```diff
@@ -0,0 +1,107 @@
+package rsrc
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/akavel/rsrc/binutil"
+	"github.com/akavel/rsrc/coff"
+	"github.com/akavel/rsrc/ico"
+	"github.com/akavel/rsrc/internal"
+)
+
+const (
+	RT_ICON       = coff.RT_ICON
+	RT_GROUP_ICON = coff.RT_GROUP_ICON
+	RT_MANIFEST   = coff.RT_MANIFEST
+)
+
+// on storing icons, see: http://blogs.msdn.com/b/oldnewthing/archive/2012/07/20/10331787.aspx
+type GRPICONDIR struct {
+	ico.ICONDIR
+	Entries []GRPICONDIRENTRY
+}
+
+func (group GRPICONDIR) Size() int64 {
+	return int64(binary.Size(group.ICONDIR) + len(group.Entries)*binary.Size(group.Entries[0]))
+}
+
+type GRPICONDIRENTRY struct {
+	ico.IconDirEntryCommon
+	Id uint16
+}
+
+func Embed(fnameout, arch, fnamein, fnameico string) error {
+	newid := make(chan uint16)
+	go func() {
+		for i := uint16(1); ; i++ {
+			newid <- i
+		}
+	}()
+
+	coff := coff.NewRSRC()
+	err := coff.Arch(arch)
+	if err != nil {
+		return err
+	}
+
+	if fnamein != "" {
+		manifest, err := binutil.SizedOpen(fnamein)
+		if err != nil {
+			return fmt.Errorf("Error opening manifest file '%s': %s", fnamein, err)
```

Comment:
hmh; given that it's now a library, the error message should now be prefixed and lowercase:

    fmt.Errorf("rsrc: error opening...